collector/lib/HostHeuristics.cpp

```diff
@@ -80,9 +80,39 @@ class DockerDesktopHeuristic : public Heuristic {
   }
 };
 
+class S390XHeuristic : public Heuristic {
+ public:
+  // S390X does not support eBPF ealier than 4.18.0-348 (rhel8.5) so we switch to use corebpf
+  // instead.
+  void Process(HostInfo& host, const CollectorConfig& config, HostConfig* hconfig) const {
+    auto k = host.GetKernelVersion();
+    std::string os_id = host.GetOSID();
+
+    if (k.machine != "s390x") {
```

Comment:
Might as well also check whether we're already configured to use core-bpf to avoid further processing